File: src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/ProtocolCodeAction.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

data class ProtocolCodeAction(
  val id: String,
  val commandID: String? = null,
  val title: String,
  val diagnostics: List<ProtocolDiagnostic>? = null,
  val kind: String? = null,
  val isPreferred: Boolean? = null,
  val disabled: DisabledParams? = null,
)

File: agent/bindings/csharp/ProtocolCodeAction.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class ProtocolCodeAction
  {
    [JsonProperty(PropertyName = "id")]
    public string Id { get; set; }
    [JsonProperty(PropertyName = "commandID")]
    public string CommandID { get; set; }
    [JsonProperty(PropertyName = "title")]
    public string Title { get; set; }
    [JsonProperty(PropertyName = "diagnostics")]
    public ProtocolDiagnostic[] Diagnostics { get; set; }
    [JsonProperty(PropertyName = "kind")]
    public string Kind { get; set; }
    [JsonProperty(PropertyName = "isPreferred")]
    public bool IsPreferred { get; set; }
    [JsonProperty(PropertyName = "disabled")]
    public DisabledParams Disabled { get; set; }
  }
}
